### DISCUSSION TITLE
Styling of tag in a styled Component.

### DISCUSSION THREAD

**Discussion by A11yship** [neuefische/questions#103] [Web] [answered]
Hello,
in my App I have the following HTML structur: 

```
<div>
<p>text</p>
<p>text</p>
</div>
```
With a styled component the div is styled and the styling of the p is nested inside. 
```
styled.div`
some styling
& p {
more styling
}`
```
Now I want to style sometimes the second p with a different color. If I use a prob and a ternary operator, change the color of both p. With a class for the second p it work. 
Is this the right solution and I have to mix styled components and css classes in this case or do you have a another suggestion?
The full code is in my [git repo](https://github.com/A11yship/capstone-project/pull/17) and you can see it on [vercel](https://capstone-project-git-digitaltimer-a11yship.vercel.app/). It is the [current task component.](https://github.com/A11yship/capstone-project/tree/DigitalTimer/src/components/CurrentTask) 

Thanks for your answer.

**Comment by ahohnsen** [comment, marked as answer]
I suggest to not nest the styles for your paragraph inside the styled component of your `div` element. You can create a styled component for the `p` element and change the color with a prop:

```jsx
const StyledParagraph = styled.p`
	padding: 5px;
	color: ${props => (props.type === 'done' ? 'green' : 'black')};
	font-size: 1.3rem;
	font-weight: bold;
`;
```

Then you need to pass that prop to your second `p` element:

```jsx
<StyledParagraph type="done">Fertig!</StyledParagraph>
```
